ocaml/quicktest/quicktest_static_vdis.ml

```diff
@@ -0,0 +1,33 @@
+
+let run_cmd cmd =
+  print_endline cmd;
+  Alcotest.(check int) cmd 0 (Sys.command cmd)
+
+let static_vdis args =
+  let static_vdis = "/opt/xensource/bin/static-vdis" in
+  let cmd = static_vdis ^ " " ^ (String.concat " " args) in
+  run_cmd cmd
+
+let test_static_vdis rpc session_id sr_info () =
+  let sr = sr_info.Qt.sr in
+  Qt.VDI.with_new rpc session_id sr (fun self ->
+      let uuid = Client.Client.VDI.get_uuid ~rpc ~session_id ~self in
+      static_vdis ["add"; uuid; "quicktest_static_vdis"];
+      static_vdis ["list"];
+      static_vdis ["attach"; uuid];
+      static_vdis ["detach"; uuid];
+      let vbds = Client.Client.VDI.get_VBDs ~rpc ~session_id ~self in
+      Alcotest.(check int) "There must be no VBDs after static-vdis detach"
+        0 (List.length vbds);
+      run_cmd ("/opt/xensource/bin/static-vdis del " ^ uuid)
+    )
+
+let tests () =
+  let open Qt_filter in
+  [ [ "test_static_vdis for SMAPIv1", `Slow, test_static_vdis ]
+    (* We need to be able to run VDI.generate_config on SMAPIv1 SRs *)
+    |> conn |> sr SR.(all |> smapiv1 |> allowed_operations [`vdi_create; `vdi_destroy] |> has_capabilities Sr_capabilities.[vdi_generate_config])
```

Comment:
I assume `vdi_create` and `vdi_destroy` are common to VDI quicktests, maybe we should have a shorthand for this? Or would it just be pointless cruft?

Reply:
I think it's better this way, because we can just look at the test list and see exactly what the tests require.